README.md

```diff
@@ -1,2 +1,1584 @@
-# services-masterclass
-A guide to developing and deploying services on a balena device
+balena Services Masterclass
+===========================
+
+A guide to developing and deploying services on a balena device.
+
+**Masterclass Type:** Core
+**Maximum Expected Time To Complete:**
+
+# Introduction
+
+balena applications consist of one or more balena services, which are based upon
+balenaEngine (a fork of Docker) images, containers, volumes and networks. This
+allows applications to be extremely flexible, allowing services to be isolated
+with access only to parts of the host required, as well as defining their own
+networking and data requirements.
+
+This masterclass acts as an introduction to how both single and multicontainer
+applications operate, as well as demonstrating some of their most common uses
+and some advanced techniques.
+
+# Hardware and Software Requirements
+
+It is assumed that the reader has access to the following:
+
+* A balena supported device, such as a [balenaFin 1.1](https://store.balena.io/collections/developer-kit/products/balenafin-v1-1-0-developer-kit),
+        [Raspberry Pi 3](https://www.raspberrypi.org/products/raspberry-pi-3-model-b/)
+        or [Intel NUC](https://www.intel.co.uk/content/www/uk/en/products/boards-kits/nuc.html). If you don't have a device, you can emulate an Intel NUC by
+        installing VirtualBox and following [this guide](https://www.balena.io/blog/no-hardware-use-virtualbox/)
+* A suitable text editor for developing code on your development platform (eg.
+    [Visual Code](https://code.visualstudio.com/))
+* A suitable shell environment for command execution (such as `bash`)
+* A [balenaCloud](https://www.balena.io/) account
+* Some familiarity with [Dockerfiles](https://docs.docker.com/engine/reference/builder/) and [docker-compose]()
+* An installation of [balena CLI](https://github.com/balena-io/balena-cli/blob/master/INSTALL.md)
+
+# Exercises
+
+All of the following exercises assume that you are able to access a suitable
+Unix based shell and that balena CLI is installed. The exercises include
```

Comment:
does this class assume having done the CLI master class first?

Reply:
Yeah, we should say that.